autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -51,15 +59,33 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         if (failed < RETRY_THRESHOLD) {
```

Comment:
The interaction of failed with the check in the else block to re-initialize the machine user - failed can end up being reset, if we see some other failures - which can potentially cause problems (specifically unnecessary delays with attempts to re-initalize a machine user). The else block forbidden failures check being delayed.

Reply:
Resolving this, and posting info on what was just discussed in a call.